packages/amplify-migration-codegen-e2e/src/helpers.ts

```diff
@@ -0,0 +1,163 @@
+import {
+  deleteProject as deleteGen1Project,
+  deleteProjectDir,
+  initJSProjectWithProfile,
+  addAuthWithDefault,
+  amplifyPushAuth,
+  getProjectMeta,
+  getUserPool,
+} from '@aws-amplify/amplify-e2e-core';
+import { npmInstall, getNpxPath, nspawn as spawn } from '@aws-amplify/amplify-e2e-core';
+import * as fs from 'fs-extra';
+import { $TSAny } from '@aws-amplify/amplify-cli-core';
+import path from 'node:path';
+import assert from 'node:assert';
+import { CloudControlClient, GetResourceCommand } from '@aws-sdk/client-cloudcontrol';
+
+const pushTimeoutMS = 1000 * 60 * 20; // 20 minutes;
+
+export async function getResourceDetails(typeName: string, identifier: string, region: string) {
+  const client = new CloudControlClient({ region });
+  const command = new GetResourceCommand({
+    TypeName: typeName,
+    Identifier: identifier,
+  });
+
+  try {
+    const response = await client.send(command);
+    return JSON.parse(response.ResourceDescription.Properties);
+  } catch (error) {
+    console.error('Error fetching resource details:', error);
+    throw error;
+  }
+}
+
+export function deployGen2CDK(cwd: string) {
+  try {
+    npmInstall(cwd);
+  } catch (error) {
+    console.error('Failed to install dependencies:', error);
+    throw error;
+  }
+  return spawn(getNpxPath(), ['ampx', 'sandbox', '--once'], {
+    cwd,
+    stripColors: true,
+    noOutputTimeout: pushTimeoutMS,
+    env: { ...process.env, npm_config_user_agent: 'npm' },
+  }).runAsync();
+}
+
+export function migrateCommand(cwd: string) {
```

Comment:
rename to `runCodegenCommand()` 
I don't see it is called outside, remove export?